doc/stdenv.xml

```diff
@@ -179,68 +179,285 @@ genericBuild
 </section>
 
 
-<section xml:id="ssec-stdenv-attributes"><title>Attributes</title>
+<section xml:id="ssec-stdenv-dependencies"><title>Specifying dependencies</title>
+
+<para>
+  As described in the Nix manual, almost any drv file store path in the a new derivation's attribute set will in fact induce a dependency on that derivation.
+  <varname>mkDerivation</varname>, however, takes a few attributes intended to, between them, include all the dependencies of a package.
+  This is done both for structure and consistency, but also so that certain other setup can take place.
+  For example, certain dependencies need their bin directories added to the <envar>PATH</envar>.
+  That is built-in, but other setup is done via a pluggable mechanism that works in conjunction with these dependency attributes.
+  See <xref linkend="ssec-setup-hooks"/> for details.
+</para>
+<para>
+  Dependencies can be broken down along three axes: their host and target platforms relative to the new derivation's, and whether they are propagated.
+  The platform distinctions are motivated by cross compilation; see <xref linkend="chap-cross"/> for exactly what each platform means.
+  <footnote><para>
+    The build platform is ignored because it is a mere implementation detail of the package satisfying the dependency:
+    As a general programming principle, dependencies are always <emphasis>specified</emphasis> as interfaces, not concrete implementation.
+  </para></footnote>
+  But even if one is not cross compiling, the platforms imply whether or not the dependency is needed at run-time or build-time, a concept that makes perfect sense outside of cross compilation.
+  For now, the run-time/build-time distinction is just a hint for mental clarity, but in the future it perhaps could be enforced.
+</para>
+<para>
+  The extension of <envar>PATH</envar> with dependencies, alluded to above, proceeds according to the relative platforms alone.
+  The process is carried only for dependencies whose host platform matches the new derivation's build platform–i.e. which runs on the directory where the new derivation will be built.
+  <footnote><para>Currently, that means all dependencies are put on the <envar>PATH</envar> for native builds, but in the future to match cross that may not be the case—the platforms would be assumed to be unique for consistency between cross and native builds.</para></footnote>
+  Of those, for each dependency <replaceable>dir</replaceable>, <filename><replaceable>dir</replaceable>/bin</filename>, if present, is added to the <envar>PATH</envar> environment variable.
```

Comment:
Perhaps replace the comma with "and". Also, `s/is added/are added/`.

Reply:
Fixed! should actually be singular in what's now a fixed "for each...".